### DISCUSSION TITLE
received netlink error: Protocol not supported

### DISCUSSION THREAD

**Discussion by timeforasong** [strongswan/strongswan#1125] [Q&A]
I am trying to establish a tunnel between two openwrt boxes(host to host). I have  following configuration. But I am getting following errors when I try to bringup particular connection.
`config setup
         uniqueids = no

conn PF_con2_0
        authby=psk
        left=192.168.125.56
        right=192.168.125.52
        auto=start
        leftupdown=/usr/libexec/adapter/VPN/ipsec_updown.sh
        keyexchange=ikev1
        rightid=192.168.125.52
        ike=sha1-aes128-modp768!
        ikelifetime=86400s
        esp=sha1-3des!
        lifetime=3600s
        lifebytes=1887436800
        margintime=30s
        marginbytes=1024000
        rightsubnet=192.168.100.0/24[ip/]
        leftsubnet=192.168.99.0/24[ip/]
`
`config setup
         uniqueids = no

conn PF_con1_0
        authby=psk
        left=192.168.125.52
        right=192.168.125.56
        auto=start
        leftupdown=/usr/libexec/adapter/VPN/ipsec_updown.sh
        keyexchange=ikev1
        rightid=192.168.125.56
        ike=sha1-aes128-modp768!
        ikelifetime=86400s
        esp=sha1-3des!
        lifetime=3600s
        lifebytes=1887436800
        margintime=30s
        marginbytes=1024000
        rightsubnet=192.168.99.0/24[ip/]
        leftsubnet=192.168.100.0/24[ip/]
`

When I run "ipsec status", it outputs following info

`Status of IKE charon daemon (strongSwan 5.9.2, Linux 4.4.60, armv7l):
  uptime: 56 minutes, since Jul 06 16:56:27 2022
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 5
  loaded plugins: charon aes des sha2 sha1 md5 random nonce x509 pubkey gmp xcbc hmac kernel-netlink socket-default connmark stroke updown eap-identity eap-md5 eap-tls whitelist unity
Listening IP addresses:
  192.168.125.56
  192.168.100.1
Connections:
   PF_con2_0:  192.168.125.56...192.168.125.52  IKEv1
   PF_con2_0:   local:  [192.168.125.56] uses pre-shared key authentication
   PF_con2_0:   remote: [192.168.125.52] uses pre-shared key authentication
   PF_con2_0:   child:  192.168.99.0/24 === 192.168.100.0/24 TUNNEL
Security Associations (1 up, 0 connecting):
   PF_con2_0[3]: ESTABLISHED 17 minutes ago, 192.168.125.56[192.168.125.56]...192.168.125.52[192.168.125.52]
   PF_con2_0[3]: IKEv1 SPIs: a234c052c18ea2b2_i* 455a991276f517ac_r, pre-shared key reauthentication in 23 hours
   PF_con2_0[3]: IKE proposal: AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_768
`
I get the following log：
`Wed Jul  6 17:36:02 2022 daemon.info : 07[CFG]   loaded IKE secret for 192.168.125.52
Wed Jul  6 17:36:02 2022 daemon.info : 10[CFG] received stroke: add connection 'PF_con2_0'
Wed Jul  6 17:36:02 2022 daemon.info : 10[CFG] added configuration 'PF_con2_0'
Wed Jul  6 17:36:02 2022 daemon.info : 08[CFG] received stroke: initiate 'PF_con2_0'
Wed Jul  6 17:36:02 2022 daemon.info : 08[IKE] initiating Main Mode IKE_SA PF_con2_0[3] to 192.168.125.52
Wed Jul  6 17:36:02 2022 authpriv.info : 08[IKE] initiating Main Mode IKE_SA PF_con2_0[3] to 192.168.125.52
Wed Jul  6 17:36:02 2022 daemon.info : 08[ENC] generating ID_PROT request 0 [ SA V V V V V ]
Wed Jul  6 17:36:02 2022 daemon.info : 08[NET] sending packet: from 192.168.125.56[500] to 192.168.125.52[500] (184 bytes)
Wed Jul  6 17:36:02 2022 user.notice VPN: syslogsource4:syslogpatternIPsec policy %s enable succeed.:syslogformatcon2
Wed Jul  6 17:36:02 2022 daemon.info : 12[NET] received packet: from 192.168.125.52[500] to 192.168.125.56[500] (164 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 12[ENC] parsed ID_PROT response 0 [ SA V V V V ]
Wed Jul  6 17:36:02 2022 daemon.info : 12[IKE] received XAuth vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 12[IKE] received DPD vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 12[IKE] received FRAGMENTATION vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 12[IKE] received NAT-T (RFC 3947) vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 12[CFG] selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_768
Wed Jul  6 17:36:02 2022 daemon.info : 12[ENC] generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
Wed Jul  6 17:36:02 2022 daemon.info : 12[NET] sending packet: from 192.168.125.56[500] to 192.168.125.52[500] (212 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 13[NET] received packet: from 192.168.125.52[500] to 192.168.125.56[500] (212 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 13[ENC] parsed ID_PROT response 0 [ KE No NAT-D NAT-D ]
Wed Jul  6 17:36:02 2022 daemon.info : 13[IKE] local host is behind NAT, sending keep alives
Wed Jul  6 17:36:02 2022 daemon.info : 13[ENC] generating ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
Wed Jul  6 17:36:02 2022 daemon.info : 13[NET] sending packet: from 192.168.125.56[4500] to 192.168.125.52[4500] (108 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 15[NET] received packet: from 192.168.125.52[4500] to 192.168.125.56[4500] (76 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 15[ENC] parsed ID_PROT response 0 [ ID HASH ]
Wed Jul  6 17:36:02 2022 daemon.info : 15[IKE] IKE_SA PF_con2_0[3] established between 192.168.125.56[192.168.125.56]...192.168.125.52[192.168.125.52]
Wed Jul  6 17:36:02 2022 authpriv.info : 15[IKE] IKE_SA PF_con2_0[3] established between 192.168.125.56[192.168.125.56]...192.168.125.52[192.168.125.52]
Wed Jul  6 17:36:02 2022 daemon.info : 15[IKE] scheduling reauthentication in 86355s
Wed Jul  6 17:36:02 2022 daemon.info : 15[IKE] maximum IKE_SA lifetime 86385s
Wed Jul  6 17:36:02 2022 daemon.info : 15[ENC] generating QUICK_MODE request 3234888510 [ HASH SA No ID ID ]
Wed Jul  6 17:36:02 2022 daemon.info : 15[NET] sending packet: from 192.168.125.56[4500] to 192.168.125.52[4500] (188 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 05[NET] received packet: from 192.168.125.52[4500] to 192.168.125.56[4500] (188 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 05[ENC] parsed QUICK_MODE response 3234888510 [ HASH SA No ID ID ]
Wed Jul  6 17:36:02 2022 daemon.info : 05[CFG] selected proposal: ESP:3DES_CBC/HMAC_SHA1_96/NO_EXT_SEQ
Wed Jul  6 17:36:02 2022 daemon.info : 05[IKE] received 1887436000 lifebytes, configured 1887436800
Wed Jul  6 17:36:02 2022 daemon.info : 05[KNL] received netlink error: Protocol not supported (93)
Wed Jul  6 17:36:02 2022 daemon.info : 05[KNL] unable to add SAD entry with SPI c7d448c8 (FAILED)
Wed Jul  6 17:36:02 2022 daemon.info : 05[KNL] received netlink error: Protocol not supported (93)
Wed Jul  6 17:36:02 2022 daemon.info : 05[KNL] unable to add SAD entry with SPI cb1d6be3 (FAILED)
Wed Jul  6 17:36:02 2022 daemon.info : 05[IKE] unable to install inbound and outbound IPsec SA (SAD) in kernel
Wed Jul  6 17:36:02 2022 daemon.info : 05[KNL] deleting policy 192.168.100.0/24 === 192.168.99.0/24 in failed, not found
Wed Jul  6 17:36:02 2022 daemon.info : 05[KNL] deleting policy 192.168.100.0/24 === 192.168.99.0/24 fwd failed, not found
Wed Jul  6 17:36:02 2022 daemon.info : 05[ENC] generating INFORMATIONAL_V1 request 1748998002 [ HASH N(NO_PROP) ]
Wed Jul  6 17:36:02 2022 daemon.info : 05[NET] sending packet: from 192.168.125.56[4500] to 192.168.125.52[4500] (76 bytes)
`
the other board log

`Wed Jul  6 17:35:58 2022 daemon.info : 08[CFG] loading secrets from '/etc/ipsec.secrets'
Wed Jul  6 17:35:58 2022 daemon.info : 08[CFG]   loaded IKE secret for 192.168.125.56
Wed Jul  6 17:35:58 2022 daemon.info : 06[CFG] received stroke: add connection 'PF_con1_0'
Wed Jul  6 17:35:58 2022 daemon.info : 06[CFG] added configuration 'PF_con1_0'
Wed Jul  6 17:35:58 2022 daemon.info : 11[CFG] received stroke: initiate 'PF_con1_0'
Wed Jul  6 17:35:58 2022 daemon.info : 11[IKE] initiating Main Mode IKE_SA PF_con1_0[3] to 192.168.125.56
Wed Jul  6 17:35:58 2022 authpriv.info : 11[IKE] initiating Main Mode IKE_SA PF_con1_0[3] to 192.168.125.56
Wed Jul  6 17:35:58 2022 daemon.info : 11[ENC] generating ID_PROT request 0 [ SA V V V V V ]
Wed Jul  6 17:35:58 2022 daemon.info : 11[NET] sending packet: from 192.168.125.52[500] to 192.168.125.56[500] (184 bytes)
Wed Jul  6 17:35:58 2022 user.notice VPN: syslogsource4:syslogpatternIPsec policy %s enable succeed.:syslogformatcon1
Wed Jul  6 17:35:58 2022 daemon.info : 10[NET] received packet: from 192.168.125.56[500] to 192.168.125.52[500] (40 bytes)
Wed Jul  6 17:35:58 2022 daemon.info : 10[ENC] parsed INFORMATIONAL_V1 request 112824380 [ N(NO_PROP) ]
Wed Jul  6 17:35:58 2022 daemon.info : 10[IKE] received NO_PROPOSAL_CHOSEN error notify
Wed Jul  6 17:35:58 2022 user.notice IpsecConnects: list index: { "lang": "zh-cn" } config: { }
Wed Jul  6 17:36:02 2022 daemon.info : 14[NET] received packet: from 192.168.125.56[449] to 192.168.125.52[500] (184 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 14[ENC] parsed ID_PROT request 0 [ SA V V V V V ]
Wed Jul  6 17:36:02 2022 daemon.info : 14[IKE] received XAuth vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 14[IKE] received DPD vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 14[IKE] received FRAGMENTATION vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 14[IKE] received NAT-T (RFC 3947) vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 14[IKE] received draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Wed Jul  6 17:36:02 2022 daemon.info : 14[IKE] 192.168.125.56 is initiating a Main Mode IKE_SA
Wed Jul  6 17:36:02 2022 authpriv.info : 14[IKE] 192.168.125.56 is initiating a Main Mode IKE_SA
Wed Jul  6 17:36:02 2022 daemon.info : 14[CFG] selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_768
Wed Jul  6 17:36:02 2022 daemon.info : 14[ENC] generating ID_PROT response 0 [ SA V V V V ]
Wed Jul  6 17:36:02 2022 daemon.info : 14[NET] sending packet: from 192.168.125.52[500] to 192.168.125.56[449] (164 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 13[NET] received packet: from 192.168.125.56[449] to 192.168.125.52[500] (212 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 13[ENC] parsed ID_PROT request 0 [ KE No NAT-D NAT-D ]
Wed Jul  6 17:36:02 2022 daemon.info : 13[IKE] remote host is behind NAT
Wed Jul  6 17:36:02 2022 daemon.info : 13[ENC] generating ID_PROT response 0 [ KE No NAT-D NAT-D ]
Wed Jul  6 17:36:02 2022 daemon.info : 13[NET] sending packet: from 192.168.125.52[500] to 192.168.125.56[449] (212 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 15[NET] received packet: from 192.168.125.56[20656] to 192.168.125.52[4500] (108 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 15[ENC] parsed ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
Wed Jul  6 17:36:02 2022 daemon.info : 15[CFG] looking for pre-shared key peer configs matching 192.168.125.52...192.168.125.56[192.168.125.56]
Wed Jul  6 17:36:02 2022 daemon.info : 15[CFG] selected peer config "PF_con1_0"
Wed Jul  6 17:36:02 2022 daemon.info : 15[IKE] IKE_SA PF_con1_0[4] established between 192.168.125.52[192.168.125.52]...192.168.125.56[192.168.125.56]
Wed Jul  6 17:36:02 2022 authpriv.info : 15[IKE] IKE_SA PF_con1_0[4] established between 192.168.125.52[192.168.125.52]...192.168.125.56[192.168.125.56]
Wed Jul  6 17:36:02 2022 daemon.info : 15[IKE] scheduling reauthentication in 86367s
Wed Jul  6 17:36:02 2022 daemon.info : 15[IKE] maximum IKE_SA lifetime 86397s
Wed Jul  6 17:36:02 2022 daemon.info : 15[ENC] generating ID_PROT response 0 [ ID HASH ]
Wed Jul  6 17:36:02 2022 daemon.info : 15[NET] sending packet: from 192.168.125.52[4500] to 192.168.125.56[20656] (76 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 01[NET] received packet: from 192.168.125.56[20656] to 192.168.125.52[4500] (188 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 01[ENC] parsed QUICK_MODE request 3234888510 [ HASH SA No ID ID ]
Wed Jul  6 17:36:02 2022 daemon.info : 01[CFG] selected proposal: ESP:3DES_CBC/HMAC_SHA1_96/NO_EXT_SEQ
Wed Jul  6 17:36:02 2022 daemon.info : 01[IKE] received 1887436000 lifebytes, configured 1887436800
Wed Jul  6 17:36:02 2022 daemon.info : 01[ENC] generating QUICK_MODE response 3234888510 [ HASH SA No ID ID ]
Wed Jul  6 17:36:02 2022 daemon.info : 01[NET] sending packet: from 192.168.125.52[4500] to 192.168.125.56[20656] (188 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 09[NET] received packet: from 192.168.125.56[20656] to 192.168.125.52[4500] (76 bytes)
Wed Jul  6 17:36:02 2022 daemon.info : 09[ENC] parsed INFORMATIONAL_V1 request 1748998002 [ HASH N(NO_PROP) ]
Wed Jul  6 17:36:02 2022 daemon.info : 09[IKE] received NO_PROPOSAL_CHOSEN error notify
`
I did install plugins as well. Is there anything else that needs to be done?

Thanks

**Comment by tobiasbrunner** [comment]
You use crappy algorithms (_modp786_, _3des_, really?) and a deprecated protocol (why would even consider using IKEv1 between two instances of strongSwan?). If you want to continue on this path, make sure the kernel has the [required modules](https://docs.strongswan.org/docs/5.9/install/kernelModules.html) loaded, which includes crypto modules (in your case _3des_ and _sha1_).

**Comment by timeforasong** [comment]
When i load all required modules,it works.
Thanks

**Reply by Thermi** [reply]
Please mark as solved
